### ISSUE TITLE
`textual` library crashing in standalone application

### ISSUE THREAD

**Issue by jimkring** [Nuitka/Nuitka#2025]
**Project to Reproduce Issue**

https://github.com/jimkring/textualize-nuitka-testing

Note that I created two different `textual` apps that crash when run after building with nuitka:

See:
- [build_ide.sh](https://github.com/jimkring/textualize-nuitka-testing/blob/main/build_ide.sh)
- [build_calculator.sh](https://github.com/jimkring/textualize-nuitka-testing/blob/main/build_calculator.sh)

These scripts are calling `--debugger --run` as follows:

```bash
#! /bin/bash
python -m nuitka \
--debugger --run \
--include-data-files=calculator/*.css=./ \
--standalone \
calculator/calculator.py
```

**Nuitka Version 1.4** 

```
$ python -m nuitka --version
1.4
Commercial: None
Python: 3.10.9 (main, Dec 15 2022, 18:18:30) [Clang 14.0.0 (clang-1400.0.29.202)]
Flavor: Homebrew Python
Executable: /Users/user/textualize-nuitka-testing/.venv/bin/python
OS: Darwin
Arch: x86_64
Version C compiler: /usr/local/opt/llvm@14/bin/clang (clang).
```

**Python Version and Environment Setup**

```
$ python3 -m venv .venv
$ source .venv/bin/activate
$ python -m pip install -r requirements.txt
```

See [requirements.txt](https://github.com/jimkring/textualize-nuitka-testing/blob/main/requirements.txt) in sample project here -> https://github.com/jimkring/textualize-nuitka-testing

* The specific PyPI names and versions

```
$ python -m pip freeze
importlib-metadata==4.13.0
markdown-it-py==2.1.0
mdurl==0.1.2
nanoid==2.0.0
Nuitka==1.4
ordered-set==4.1.0
Pygments==2.14.0
rich==13.3.1
textual==0.10.1
zipp==3.12.0
```

**Run with Debugging**

```
(lldb) run
Assertion failed: (PyList_Check(tmp_list_arg_value_1)), function impl_rich$box$$$function__8_get_bottom, file module.rich.box.c, line 6183.
calculator.bin was compiled with optimization - stepping may behave oddly; variables may not be available.
Process 18566 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = hit program assert
    frame #4: 0x0000000100711405 calculator.bin`impl_rich$box$$$function__8_get_bottom(self=<unavailable>, python_pars=<unavailable>) at module.rich.box.c:6183:9 [opt]
   6180             type_description_1 = "oooooo";
   6181             goto frame_exception_exit_1;
   6182         }
-> 6183         assert(PyList_Check(tmp_list_arg_value_1));
   6184         tmp_result = LIST_APPEND1(tmp_list_arg_value_1, tmp_item_value_1);
   6185         assert(!(tmp_result == false));
   6186         tmp_unused = NUITKA_VOID_OK;
Process 18566 launched: '/Users/user/textualize-nuitka-testing/calculator.dist/calculator.bin' (x86_64)
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = hit program assert
    frame #0: 0x00007ff80433900e libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x00007ff80436f1ff libsystem_pthread.dylib`pthread_kill + 263
    frame #2: 0x00007ff8042bad24 libsystem_c.dylib`abort + 123
    frame #3: 0x00007ff8042ba0cb libsystem_c.dylib`__assert_rtn + 314
  * frame #4: 0x0000000100711405 calculator.bin`impl_rich$box$$$function__8_get_bottom(self=<unavailable>, python_pars=<unavailable>) at module.rich.box.c:6183:9 [opt]
    frame #5: 0x0000000101a6759c calculator.bin`CALL_FUNCTION_WITH_SINGLE_ARG(called=0x0000000107735350, arg=0x00000001079ac4c0) at HelpersCalling2.c:435:26 [opt]
    frame #6: 0x0000000100a1ad9d calculator.bin`rich$panel$$$function__5___rich_console__$$$genobj__1___rich_console___context(generator=0x0000000106054810, yield_return_value=0x0000000102fe29d0) at module.rich.panel.c:4822:37 [opt]
    frame #7: 0x0000000101a50592 calculator.bin`_Nuitka_Generator_send(generator=0x0000000106054810, value=0x0000000102fe29d0, exception_type=<unavailable>, exception_value=<unavailable>, exception_tb=<unavailable>) at CompiledGeneratorType.c:587:23 [opt]
    frame #8: 0x0000000101a51207 calculator.bin`Nuitka_Generator_tp_iternext(generator=<unavailable>) at CompiledGeneratorType.c:793:12 [opt]
    frame #9: 0x000000010081c633 calculator.bin`rich$console$$$function__82_render$$$genobj__1_render_context [inlined] ITERATOR_NEXT(iterator=<unavailable>) at iterators.h:204:24 [opt]
    frame #10: 0x000000010081c61d calculator.bin`rich$console$$$function__82_render$$$genobj__1_render_context(generator=0x0000000106050a10, yield_return_value=<unavailable>) at module.rich.console.c:29313:32 [opt]
    frame #11: 0x0000000101a50592 calculator.bin`_Nuitka_Generator_send(generator=0x0000000106050a10, value=0x0000000102fe29d0, exception_type=<unavailable>, exception_value=<unavailable>, exception_tb=<unavailable>) at CompiledGeneratorType.c:587:23 [opt]
    frame #12: 0x0000000101a51207 calculator.bin`Nuitka_Generator_tp_iternext(generator=<unavailable>) at CompiledGeneratorType.c:793:12 [opt]
    frame #13: 0x0000000101af5b65 calculator.bin`Nuitka_YieldFromGeneratorCore [inlined] _Nuitka_YieldFromGeneratorCore(generator=0x0000000106051a10, yieldfrom=0x0000000106050a10, send_value=0x0000000102fe29d0) at CompiledGeneratorType.c:321:18 [opt]
    frame #14: 0x0000000101af5b50 calculator.bin`Nuitka_YieldFromGeneratorCore(generator=0x0000000106051a10, send_value=0x0000000102fe29d0) at CompiledGeneratorType.c:363:25 [opt]
    frame #15: 0x0000000101a5058c calculator.bin`_Nuitka_Generator_send [inlined] Nuitka_YieldFromGeneratorInitial(generator=0x0000000106051a10, send_value=0x0000000102fe29d0) at CompiledGeneratorType.c:409:24 [opt]
    frame #16: 0x0000000101a50587 calculator.bin`_Nuitka_Generator_send(generator=0x0000000106051a10, value=0x0000000102fe29d0, exception_type=<unavailable>, exception_value=<unavailable>, exception_tb=<unavailable>) at CompiledGeneratorType.c:589:23 [opt]
    frame #17: 0x0000000101a51207 calculator.bin`Nuitka_Generator_tp_iternext(generator=<unavailable>) at CompiledGeneratorType.c:793:12 [opt]
    frame #18: 0x0000000101a73bdd calculator.bin`LIST_EXTEND(target=0x000000010792a040, other=0x0000000106051a10) at HelpersLists.c:262:26 [opt]
    frame #19: 0x0000000101a757a4 calculator.bin`MAKE_LIST(iterable=0x0000000106051a10) at HelpersLists.c:698:16 [opt]
    frame #20: 0x0000000101019185 calculator.bin`impl_textual$app$$$function__65_panic$$$function__1_render(self=0x00000001078f6c40, python_pars=<unavailable>) at module.textual.app.c:34200:31 [opt]
    frame #21: 0x0000000101a67551 calculator.bin`CALL_FUNCTION_WITH_SINGLE_ARG(called=0x00000001078f6c40, arg=0x0000000107923a00) at HelpersCalling2.c:376:22 [opt]
    frame #22: 0x0000000101016db4 calculator.bin`impl_textual$app$$$function__65_panic(self=<unavailable>, python_pars=<unavailable>) at module.textual.app.c:33348:40 [opt]
    frame #23: 0x0000000101a4e90f calculator.bin`Nuitka_CallMethodFunctionPosArgs(function=0x0000000107754220, object=0x0000000107914c40, args=0x00007ff7bfefc680, args_size=1) at CompiledFunctionType.c:2324:12 [opt]
    frame #24: 0x0000000101a920bd calculator.bin`CALL_METHOD_WITH_SINGLE_ARG(source=0x0000000107914c40, attr_name=<unavailable>, arg=0x0000000107923a00) at HelpersCalling2.c:10680:21 [opt]
    frame #25: 0x0000000101019cae calculator.bin`impl_textual$app$$$function__66__handle_exception(self=<unavailable>, python_pars=<unavailable>) at module.textual.app.c:34396:29 [opt]
    frame #26: 0x0000000101a4e90f calculator.bin`Nuitka_CallMethodFunctionPosArgs(function=0x0000000107754310, object=0x0000000107914c40, args=0x00007ff7bfefc790, args_size=1) at CompiledFunctionType.c:2324:12 [opt]
    frame #27: 0x0000000101a920bd calculator.bin`CALL_METHOD_WITH_SINGLE_ARG(source=0x0000000107914c40, attr_name=<unavailable>, arg=0x0000000107923a00) at HelpersCalling2.c:10680:21 [opt]
    frame #28: 0x00000001010296e9 calculator.bin`textual$app$$$function__69__process_messages$$$coroutine__1__process_messages_context(coroutine=0x000000010604f410, yield_return_value=<unavailable>) at module.textual.app.c:40760:30 [opt]
    frame #29: 0x0000000101af9469 calculator.bin`_Nuitka_YieldFromCoroutineCore(coroutine=0x000000010604f410, send_value=0x0000000102fe29d0, mode=<unavailable>) at CompiledCoroutineType.c:265:19 [opt]
    frame #30: 0x0000000101af8dfa calculator.bin`_Nuitka_Coroutine_sendR [inlined] Nuitka_YieldFromCoroutineNext(coroutine=0x000000010604f410) at CompiledCoroutineType.c:316:24 [opt]
    frame #31: 0x0000000101af8ddc calculator.bin`_Nuitka_Coroutine_sendR(coroutine=0x000000010604f410, value=<unavailable>, closing=<unavailable>, exception_type=<unavailable>, exception_value=<unavailable>, exception_tb=<unavailable>, result=0x00007ff7bfefc9a0) at CompiledCoroutineType.c:441:23 [opt]
    frame #32: 0x0000000101af8956 calculator.bin`_Nuitka_Coroutine_send(coroutine=<unavailable>, value=<unavailable>, closing=<unavailable>, exception_type=0x0000000000000000, exception_value=0x0000000000000000, exception_tb=0x0000000000000000) at CompiledCoroutineType.c:595:9 [opt]
    frame #33: 0x0000000101af97b8 calculator.bin`_Nuitka_YieldFromCore(yieldfrom=0x0000000107917220, send_value=0x0000000102fe29d0, returned_value=0x00007ff7bfefc9f8, mode=<unavailable>) at CompiledCoroutineType.c:208:18 [opt]
    frame #34: 0x0000000101af941e calculator.bin`_Nuitka_YieldFromCoroutineCore(coroutine=0x0000000106043a10, send_value=0x0000000102fe29d0, mode=<unavailable>) at CompiledCoroutineType.c:259:25 [opt]
    frame #35: 0x0000000101af8dfa calculator.bin`_Nuitka_Coroutine_sendR [inlined] Nuitka_YieldFromCoroutineNext(coroutine=0x0000000106043a10) at CompiledCoroutineType.c:316:24 [opt]
    frame #36: 0x0000000101af8ddc calculator.bin`_Nuitka_Coroutine_sendR(coroutine=0x0000000106043a10, value=<unavailable>, closing=<unavailable>, exception_type=<unavailable>, exception_value=<unavailable>, exception_tb=<unavailable>, result=0x00007ff7bfefcaa0) at CompiledCoroutineType.c:441:23 [opt]
    frame #37: 0x0000000101af8956 calculator.bin`_Nuitka_Coroutine_send(coroutine=<unavailable>, value=<unavailable>, closing=<unavailable>, exception_type=0x0000000000000000, exception_value=0x0000000000000000, exception_tb=0x0000000000000000) at CompiledCoroutineType.c:595:9 [opt]
    frame #38: 0x0000000101af97b8 calculator.bin`_Nuitka_YieldFromCore(yieldfrom=0x0000000107916ef0, send_value=0x0000000102fe29d0, returned_value=0x00007ff7bfefcaf8, mode=<unavailable>) at CompiledCoroutineType.c:208:18 [opt]
    frame #39: 0x0000000101af941e calculator.bin`_Nuitka_YieldFromCoroutineCore(coroutine=0x000000010487bc10, send_value=0x0000000102fe29d0, mode=<unavailable>) at CompiledCoroutineType.c:259:25 [opt]
    frame #40: 0x0000000101af8dfa calculator.bin`_Nuitka_Coroutine_sendR [inlined] Nuitka_YieldFromCoroutineNext(coroutine=0x000000010487bc10) at CompiledCoroutineType.c:316:24 [opt]
    frame #41: 0x0000000101af8ddc calculator.bin`_Nuitka_Coroutine_sendR(coroutine=0x000000010487bc10, value=<unavailable>, closing=<unavailable>, exception_type=<unavailable>, exception_value=<unavailable>, exception_tb=<unavailable>, result=0x00007ff7bfefcbc8) at CompiledCoroutineType.c:441:23 [opt]
    frame #42: 0x0000000101afbefb calculator.bin`_Nuitka_Coroutine_amsend(coroutine=<unavailable>, arg=<unavailable>, result=<unavailable>) at CompiledCoroutineType.c:1046:24 [opt]
    frame #43: 0x0000000106e6ace2 _asyncio.so`task_step_impl + 427
    frame #44: 0x0000000106e6aac8 _asyncio.so`task_step + 48
    frame #45: 0x0000000102d41a23 Python`_PyObject_MakeTpCall + 132
    frame #46: 0x0000000102e53cdb Python`context_run + 81
    frame #47: 0x0000000102d8fe6c Python`cfunction_vectorcall_FASTCALL_KEYWORDS + 89
    frame #48: 0x0000000102e2e230 Python`_PyEval_EvalFrameDefault + 3181
    frame #49: 0x0000000102e2c2f7 Python`_PyEval_Vector + 342
    frame #50: 0x0000000102e3c283 Python`call_function + 172
    frame #51: 0x0000000102e3335c Python`_PyEval_EvalFrameDefault + 23961
    frame #52: 0x0000000102e2c2f7 Python`_PyEval_Vector + 342
    frame #53: 0x0000000102e3c283 Python`call_function + 172
    frame #54: 0x0000000102e3335c Python`_PyEval_EvalFrameDefault + 23961
    frame #55: 0x0000000102e2c2f7 Python`_PyEval_Vector + 342
    frame #56: 0x0000000102e3c283 Python`call_function + 172
    frame #57: 0x0000000102e3335c Python`_PyEval_EvalFrameDefault + 23961
    frame #58: 0x0000000102e2c2f7 Python`_PyEval_Vector + 342
    frame #59: 0x0000000102e3c283 Python`call_function + 172
    frame #60: 0x0000000102e3335c Python`_PyEval_EvalFrameDefault + 23961
    frame #61: 0x0000000102e2c2f7 Python`_PyEval_Vector + 342
    frame #62: 0x0000000102e3cc2c Python`PyEval_EvalCodeEx + 460
    frame #63: 0x0000000101a7e59e calculator.bin`callPythonFunction(func=0x0000000106dfc550, args=0x00007ff7bfefdd28, count=1) at HelpersCalling.c:76:24 [opt]
    frame #64: 0x0000000101a6769f calculator.bin`CALL_FUNCTION_WITH_SINGLE_ARG(called=0x0000000106dfc550, arg=0x000000010487bc10) at HelpersCalling2.c:539:16 [opt]
    frame #65: 0x0000000100ff9cd8 calculator.bin`impl_textual$app$$$function__41_run(self=<unavailable>, python_pars=<unavailable>) at module.textual.app.c:22514:29 [opt]
    frame #66: 0x0000000101a4e3fc calculator.bin`Nuitka_CallMethodFunctionNoArgs(function=0x0000000107742f20, object=0x0000000107914c40) at CompiledFunctionType.c:2312:12 [opt]
    frame #67: 0x0000000101a91a80 calculator.bin`CALL_METHOD_NO_ARGS(source=0x0000000107914c40, attr_name=0x00000001030ce930) at HelpersCalling2.c:10455:36 [opt]
    frame #68: 0x000000010001c163 calculator.bin`modulecode___main__(module=<unavailable>, loader_entry=<unavailable>) at module.__main__.c:15980:29 [opt]
    frame #69: 0x0000000101af1148 calculator.bin`_EXECUTE_EMBEDDED_MODULE at MetaPathBasedLoader.c:1026:28 [opt]
    frame #70: 0x0000000101af1106 calculator.bin`_EXECUTE_EMBEDDED_MODULE(module=0x0000000103635e40, module_name=0x0000000103663a30, name="__main__") at MetaPathBasedLoader.c:1065:18 [opt]
    frame #71: 0x0000000101af0f81 calculator.bin`IMPORT_EMBEDDED_MODULE(name="__main__") at MetaPathBasedLoader.c:1119:24 [opt]
    frame #72: 0x0000000101a4a979 calculator.bin`EXECUTE_MAIN_MODULE(module_name="__main__") at MainProgram.c:250:12 [opt]
    frame #73: 0x0000000101a4a7f8 calculator.bin`main(argc=1, argv=0x00007ff7bfefede8) at MainProgram.c:1129:5 [opt]
    frame #74: 0x00000001027b152e dyld`start + 462
(lldb) quit
```

**Comment by kayhayen** [comment]
This is a regression of 1.4 where the list attributes when called through a variable, to not work on "__self__" of that method, but rather on the method itself, causing this assertion. It will be fixed in 1.4.2 of course.

Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html

**Comment by kayhayen** [comment]
Part of the 1.4.2 hotfix I just made.
